httpsig: add RFC 9421 HTTP Message Signatures support (#21239)
curl/curl: docs/cmdline-opts/httpsig-algorithm.md

---
c: Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
SPDX-License-Identifier: curl
Long: httpsig-algorithm
Protocols: HTTP
Arg: <algorithm>
Help: Algorithm for HTTP Message Signatures
Category: auth http
Added: 8.21.0
Multi: single
Experimental: yes
See-also:
  - httpsig-key
  - httpsig-keyid
  - httpsig-headers
Example:
  - --httpsig-algorithm ed25519 --httpsig-key key.hex --httpsig-keyid "my-key" $URL
  - --httpsig-algorithm hmac-sha256 --httpsig-key secret.hex --httpsig-keyid "shared" $URL
---

> # `--httpsig-algorithm`

Reviewer: Maybe just `--httpsig-algo` to shorten it a little?

Sign outgoing HTTP requests using RFC 9421 HTTP Message Signatures.

> The algorithm argument specifies which signing algorithm to use. Supported

Reviewer: We call them command line options:

```suggestion
This option specifies which signing algorithm to use. Supported
```

values are **ed25519** and **hmac-sha256**. Any other value causes curl to

> exit with an error.

Reviewer: What about setting one (ed25519 maybe?) as a default so that this option isn't required for all uses?

Author: Done
